xGame/Java/xgame/src/main/java/com/xgame/restservice/controller/UserController.java

```diff
@@ -44,8 +45,14 @@ public UserViewModel login(@RequestBody UserCredentials credentials) {
 		return userService.login(credentials);
 	}
 	
+
 	@PostMapping("/user/deactivate")
 	public void deactivateUser(@RequestBody Integer id) {
 		userService.deactivateUser(id);
 	}
+	
+	@GetMapping("/user/profile")
+	public ProfileViewModel getProfile(@RequestBody Integer id) {
```

Comment:
you don't need to use the request body. you shouldn't really use a body in a GET either. Use instead:
`@RequestParam(value = "id") int id`
as the parameter.